app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -390,8 +374,40 @@ public function buildBoltCheckoutJavascript($checkoutType, $quote, $hintData, $c
                   close: function() {
                      $onCloseCallback
                   }
+                }   
+            );
+        ";
+
+        return
+        ("
+            var \$hints_transform = $hintsTransformFunction;
+            
+            var json_cart = $jsonCart;
+            var json_hints = \$hints_transform($jsonHints);
+            var order_completed = false;
+            var check_count = 0;
+            var configure_bolt = function() {
+                $boltConfigureCall
+                return true;
+            };
+            
+            BoltCheckout.configure(
```

Comment:
@daisy1754 I have full confidence that it works as I tested both.  However, I did implement this in an alternative manner that I passed over to avoid conversations about "hacky" approaches.  *I will use the alternative approach, but would advice that the Bolt code not change so that configure could be called from within check as it can be now.  This is very powerful as is.*

Monitoring a blur event is the most expensive event to be monitoring and would invoke several blocking modals decreasing UX in both firecheckout and the admin at minimum.  In worse case, we also incur several http ajax calls and possible create several useless quotes.  This approach should be avoided when possible and it is proven possible to avoid.  If we want to check the state without this extra overhead and not rely on the natural event, **Bolt button click**, with the current design, we should use the initial approach polling approach.

Still, the most reasonable present and/or future alternatives for event driven efficient Bolt button configuration are:

1.) Bolt provides an click event that is performed prior to opening the modal that provides the opportunity to run or re-run configure.  Preferably, configure is not run until click.  In other words, separate the idea of Bolt configuration from that of Bolt Button render.  They are not the same and should not be treated in the same way.  Or if we want to use different terms to think about this more clearly, we can say that we first `configure` the Bolt button and once it is configured, we `render` the button.  After the rendered button is clicked, we `check` the button to if we can `load` the bolt form.  Loading options set by configure can be change at this time prior to the bolt form loading.

2.) We accept the fact that the `check` function allows for configure to be recalled and we make sure to respect this and ensure that it remains supported in the future.  This is probably the best approach that only requires the addition of some other unit and integration test for the Bolt modal.  It also makes sense that in the Bolt `check` method, be perform

Reply:
Both your suggestion and current implementation initialize token only after user clicks the button. Given "instant load" is a huge value prop for us (https://mailchi.mp/319511d27567/new-at-bolt-instant-load-for-bolt-checkout-and-address-validation-2767153?utm_source=Testers&utm_campaign=284d8426c8-EMAIL_CAMPAIGN_2018_12_27_07_26_COPY_02&utm_medium=email&utm_term=0_fb951463cd-284d8426c8-517057421 ) I believe we should call `configure` whenever form is ready. Please note configure call does not invoke any blocking logic or UI.

Whenever form is updated, we should check if form is ready. Then we can call configure, which will involve Ajax call. I'd assume for most of cases you usually don't edit form much after it's complete but I might miss something here.

Reply:
I now understand the priority is for UX and instant load is highly valued.  I certainly will correct standard checkout and payment only checkout with your specification of calling BoltCheckout.configure immediately.

Regarding Firecheckout and admin checkout, we are not concerned with `BoltCheckout.configure`'s blocking logic, but rather the blocking logic of the UI incurred by calling the native `validate` methods of FireCheckout and of the admin form.  In both cases, the UI is blocked and we are trying to avoid making additional calls to this method.  However, I can create a solution for this without having to make additional calls and honoring the goal of "instant load".

Reply:
hmm I don't see need for calling `validate` before calling BoltCheckout.configure. You can call configure when order is ready from Bolt's standpoint.

How about following:
 - we resister listener to detect change in fields that Bolt cares about (eg., email, name, or street address)
 - when listener is called, we check if all required fields from Bolt perspective is ready
 - if so we call BoltCheckout.configure with promise (this will mean Ajax call)
 - when user actually click checkout button, within check callback, we call framework's validate method

Reply:
The need for not creating 1000s of extra quotes in the context of Product Page loads coupled with the minimal difference in time for Bolt to appear outweigh any minute gains in instant loan in the Product Page and Firecheckout context.  Instant load is not a priority in admin order creation.  Instant load will be respected in the shopping cart and the one-page-checkout scenario by calling BoltCheckout.configure immediately.